pkg/steps/input_image_tag.go

```diff
@@ -72,14 +75,44 @@ func (s *inputImageTagStep) Run(ctx context.Context) error {
 }
 
 func (s *inputImageTagStep) run(ctx context.Context) error {
-	logrus.Infof("Tagging %s into %s:%s.", s.config.BaseImage.ISTagName(), api.PipelineImageStream, s.config.To)
-
 	if _, err := s.Inputs(); err != nil {
 		return fmt.Errorf("could not resolve inputs for image tag step: %w", err)
 	}
+
+	var objectReferenceName string
+	if s.config.ExternalImage != nil {
+		logrus.Infof("Tagging %s into %s:%s.", externalImageReference(s.config), api.PipelineImageStream, s.config.To)
+		objectReferenceName = externalImageReference(s.config)
+
+		if s.config.ExternalImage.PullSecret != "" {
+			// Get the secret from the test-credentials namespace. We expect it exists
+			ciSecret := &coreapi.Secret{}
+			err := s.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: "test-credentials", Name: s.config.ExternalImage.PullSecret}, ciSecret)
```

Comment:
Is this the right place to grab the secret? ci-operator has already the mechanisms of gathering any secret is required for the runtime. Perhaps it would be better to do it in one place and pass the secret to the steps.